sdk/core/core/src/az_curl_adapter.c

```diff
@@ -229,7 +229,7 @@ AZ_NODISCARD az_result setup_url(CURL * const p_curl, az_http_request_builder co
   }
 
   // free used buffer before anything else
-  az_mut_span_set(writable_buffer, 0);
+  AZ_IGNORE_RESULT(az_mut_span_memset(writable_buffer, 0));
```

Comment:
IMHO, we should either check the result or change the signature to `void az_mut_span_memset`

Reply:
I don't think so. There could be other contexts when it could be important to know that the function call is succeeded, and while some part of me want elegant code, some other part of me feels that making it void could be similar to adjusting the results to the desired outcome.

Reply:
Ok for now, I think we need to discuss pure functions, or non-throwable functions and default initialization to make our code more deterministic.